extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -245,6 +251,42 @@ export class ResourceTypeService implements IResourceTypeService {
 		return undefined;
 	}
 
+	/**
+	 * Get the ok button text based on the selected options
+	 */
+	private getOkButtonText(resourceType: ResourceType, selectedOptions: { option: string, value: string }[]): string | undefined {
+		if (resourceType.okButtonText) {
+			for (let i = 0; i < resourceType.okButtonText.length; i++) {
+				const buttonText = resourceType.okButtonText[i];
+				if (buttonText.when === undefined || buttonText.when.toString().toLowerCase() === 'true') {
+					return buttonText.value;
+				} else {
+					let expected = buttonText.when.replace(' ', '').split('&&').sort();
+
+					let actual: string[] = [];
+					selectedOptions.forEach(option => {
+						actual.push(`${option.option}=${option.value}`);
+					});
+					actual = actual.sort();
+
+					if (actual.length === expected.length) {
```

Comment:
If I'm reading this correctly - you're requiring that if someone uses a when clause they have to have every single option type included? What if there's 3 options on the page but I only care about switching the text based on the value of one of them?

Reply:
I currently have it that each text has its own when clause, and that ones that have to share the same text, have to be separate in the text-clause array. I based this code on getProvider, so I will need to change it to better fit the task.